modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/Swift5ClientCodegen.java

```diff
@@ -572,6 +585,12 @@ public void processOpts() {
         supportingFiles.add(new SupportingFile("Extensions.mustache",
                 sourceFolder,
                 "Extensions.swift"));
+        supportingFiles.add(new SupportingFile("OpenISO8601DateFormatter.mustache",
+                sourceFolder,
+                "OpenISO8601DateFormatter.swift"));
+        supportingFiles.add(new SupportingFile("DateWithoutTime.mustache",
+                sourceFolder,
+                "DateWithoutTime.swift"));
```

Comment:
@Jonas1893 I think this file should be optional, right?
I'm not sure if we should give it a more specific name like `OpenAPIDateWithoutTime`.
What do you think?

Reply:
Regarding point 1: I think that works. Then I would also need to move the JSONEncoding conformance out of the Extensions file and break convention here. But is probably better than generating classes that are not needed
2. Good suggestion. If we would need a custom type for both date and date-time I would go with OpenAPIDate instead but since we don't I like your suggestion. Let's be as precise as possible.

Reply:
@4brunu changed